Find a functionally equivalent code:android.widget.Switch.stopDrag:COMMENT
Method Modifier: private     
Comment:/**
 * Called from onTouchEvent to end a drag operation.
 *
 * @param ev Event that triggered the end of drag mode - ACTION_UP or ACTION_CANCEL
 */

@@ -1,19 +1,20 @@
 {
     mTouchMode = TOUCH_MODE_IDLE;
-    // Up and not canceled, also checks the switch has not been disabled during the drag
-    boolean commitChange = ev.getAction() == MotionEvent.ACTION_UP && isEnabled();
-    cancelSuperTouch(ev);
+    // Commit the change if the event is up and not canceled and the switch
+    // has not been disabled during the drag.
+    final boolean commitChange = ev.getAction() == MotionEvent.ACTION_UP && isEnabled();
+    final boolean newState;
     if (commitChange) {
-        boolean newState;
         mVelocityTracker.computeCurrentVelocity(1000);
-        float xvel = mVelocityTracker.getXVelocity();
+        final float xvel = mVelocityTracker.getXVelocity();
         if (Math.abs(xvel) > mMinFlingVelocity) {
             newState = isLayoutRtl() ? (xvel < 0) : (xvel > 0);
         } else {
             newState = getTargetCheckedState();
         }
-        animateThumbToCheckedState(newState);
     } else {
-        animateThumbToCheckedState(isChecked());
+        newState = isChecked();
     }
+    setChecked(newState);
+    cancelSuperTouch(ev);
 }

